CMakeLists.txt

```diff
@@ -46,6 +46,14 @@ if(BUILD_TESTING)
 
   ament_lint_auto_find_test_dependencies()
 
+  ament_add_gtest(test_asserts_ndebug test/test_asserts.cpp)
+
+  if(TARGET test_asserts_ndebug)
+    add_definitions(-DNDEBUG)
```

Comment:
Same for windows?

Reply:
It looks like it is since NDEBUG is defined in the standard library.

Reply:
Maybe https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html? As written this will affect the whole package. Which might be kind of strange, since you'd expect `NDEBUG` to be missing from `librcpputils` if you built with `-DCMAKE_BUILD_TYPE=Release`, but if this test is also being built that would not be the case any longer. Since the thing being tested is an inlined function in a header only, it should be sufficient to just build the test executable's target with `-DNDEBUG`, right?